pkg/app/produce.go

```diff
@@ -0,0 +1,220 @@
+package app
+
+import (
+	"context"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/go-kit/kit/log/level"
+	"github.com/gogo/protobuf/proto"
+	"github.com/golang/snappy"
+	"github.com/prometheus/common/model"
+	"github.com/prometheus/prometheus/prompb"
+
+	mcontext "github.com/grafana/prometheus-pulsar-remote-write/pkg/context"
+	"github.com/grafana/prometheus-pulsar-remote-write/pkg/pulsar"
+)
+
+const errSendingSamples = "Error sending samples to remote storage"
+
+type produceCommand struct {
+	app  *App
+	name string
+
+	writePath     string
+	replicaLabels []string
+	metrics       *metrics
+	sendTimeout   time.Duration
+	pulsar        *pulsarConfig
+}
+
+func newProduceCommand(app *App) *produceCommand {
+	name := "produce"
+	p := &produceCommand{
+		name:    name,
+		app:     app,
+		metrics: newMetrics(app.registry),
+		pulsar:  &pulsarConfig{},
+	}
+
+	cmd := app.app.Command(name, "Receive remote_write requests and produce messages on the pulsar bus").Default()
```

Comment:
>The default and only mode is produce, which should not break any existing users.

When you set a command as `Default()` will it be the triggered command even if it's not provided?

Reply:
Yes exactly apart from `--help`, both commands behave exactly the same way

```
./prometheus-pulsar-remote-write --x y
./prometheus-pulsar-remote-write produce --x y
```

```
# will fail
./prometheus-pulsar-remote-write invalid 

# will run consume
./prometheus-pulsar-remote-write consume
```